### PR TITLE
Convert relative to absolute dates

### PR DESCRIPTION
**Pull request by domingohui**
#91

### CODE REVIEW

File: internal_displacement/interpreter.py [hunk 1]
```diff
@@ -365,17 +395,6 @@ def match_entities_in_block(self, entities, token_block):
                 matched.append(e)
         return matched
 
-    def convert_to_facts(self, fact_array, fact_type, start_offset=0):
```
Comment (simonb83) [inline internal_displacement/interpreter.py]: I'm not sure why you're removing this function?

File: internal_displacement/interpreter.py [hunk 2]
```diff
@@ -413,18 +434,23 @@ def extract_locations(self, sentence, root=None):
         else:
             return []
 
-    def extract_dates(self, sentence, story, root=None):
+    def extract_dates(self, sentence, story, publication_date=None, root=None):
```
Comment (simonb83) [inline internal_displacement/interpreter.py]: I have a feeling we can simplify this function a bit more. I did some testing and found I actually seemed to get better results by just using DATE entities extracted from the sentence using Spacy. What do you think?

File: internal_displacement/interpreter.py [hunk 3]
```diff
@@ -443,37 +469,44 @@ def extract_dates(self, sentence, story, root=None):
             block_dates = self.match_entities_in_block(
                 date_entities, contiguous_token_block)
 
-            likely_block_dates = self.date_likelihood(block_dates, story)
+            # Filter unlikely dates from possible dates
+            likely_block_dates = self.date_likelihood(block_dates, publication_date)
             if likely_block_dates:
-                return self.convert_to_facts(likely_block_dates, "date", sentence[0].idx)
+                return likely_block_dates
```
Comment (simonb83) [inline internal_displacement/interpreter.py]: The reason for the returning extracted entities as Facts is to have a wrapper that can keep track of multiple things at the same time:
1. The underlying text
2. The token or tokens associated with that text
3. The location of the tokens within the text

File: internal_displacement/interpreter.py [hunk 4]
```diff
@@ -755,31 +791,28 @@ def cleanup(self, text):
...
         ----------
-        story:      the article content:String
+        story:                  the article content:String
+        publication_date:       the publication date of the article:datetime
         """
         story = self.cleanup(story)
         processed_reports = []
         story = self.nlp(story)
         sentences = list(story.sents)  # Split into sentences
-        dates_memory = []  # Keep a running track of the most recent dates found in articles
```
Comment (simonb83) [inline internal_displacement/interpreter.py]: Why are we no longer keeping track of previously encountered dates?

File: internal_displacement/interpreter.py (hunk 3 above)
Comment (domingohui) [inline internal_displacement/interpreter.py]: I see. I just didn't see Fact being used elsewhere. I will put it back in.

**Pull request merged**